### PR TITLE
fix metric logging error and support specified device in convert device

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
1) fixed the error when logging a vector-valued metric
2) add an optional argument to the ``conver_device`` function for specifying target device
3) this is a PR created from a forked branch which tests CI support for external PRs at the same time

### CODE REVIEW

File: alf/nest/utils.py [hunk 1]
```diff
@@ -207,10 +221,14 @@ def _convert_cpu(tensor):
...
-    d = alf.get_default_device()
+    if device is None:
+        d = alf.get_default_device()
+    else:
+        d = device
+
     if d == 'cpu':
         return nest.map_structure(_convert_cpu, nests)
-    elif d == 'cuda':
+    elif d == 'cuda' and torch.cuda.is_available():
```
Comment (hnyu) [inline alf/nest/utils.py]: cuda.is_available() should be inside this branch and raise error otherwise

File: alf/trainers/policy_trainer.py [hunk 2]
```diff
@@ -694,7 +695,8 @@ def play(root_dir,
             time_step = env.reset()
 
     for m in metrics:
-        logging.info("%s: %f", m.name, m.result())
+        logging.info("%s: %s", m.name,
+                     map_structure(lambda x: x.numpy().item(), m.result()))
```
Comment (hnyu) [inline alf/trainers/policy_trainer.py]: Do we need item()? It could be an np.array for each element?

**Review by hnyu — CHANGES_REQUESTED** [review]

File: alf/nest/utils.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/nest/utils.py]: Done

File: alf/trainers/policy_trainer.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/trainers/policy_trainer.py]: ``item()`` is needed, as the ``AverageReturnMetric`` returns a list of ndim-0 tensors (one for each return/reward dimension) in the case of play. Without ``item()``, we will get something like:
```
 AverageReturn: [array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32)]
```
By adding item, we get a cleaner log message like:
```
AverageReturn: [0.0, 0.0, 0.0, 0.0, 0.0]
```



Added support to the case of np.array for each element, although currently there is no such metrics in play.

**Review by hnyu — APPROVED** [review]

**Pull request merged**